src/App.tsx

```diff
@@ -1,223 +1,150 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+/* eslint-disable no-console */
+import React, {
+  // FormEvent,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+
+import { Todo } from './types/Todo';
+import { deleteTodo, getTodos, postTodo } from './api/todos';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterValue, setFilterValue] = useState('all');
+  const [title, setTitle] = useState('');
+  const [isTodoAdded, setIsTodoAdded] = useState(false);
+  const [selectedTodosId, setSelectedTodosId] = useState<number[]>([]);
+
   const newTodoField = useRef<HTMLInputElement>(null);
 
-  useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+  const focusOnInput = () => {
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+  };
+
+  useEffect(() => {
+    focusOnInput();
+
+    getTodos(user?.id || 0)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+      });
   }, []);
 
+  const hundleAddTodo = async (inputTitle: string) => {
+    setTitle(inputTitle);
+    setIsTodoAdded(true);
+
+    try {
+      if (user) {
+        const newTodo = await postTodo(user.id, inputTitle);
+
+        setTodos([...todos, newTodo]);
+      }
+    } catch {
+      setErrorMessage('Unable to add todo');
+    } finally {
+      setIsTodoAdded(false);
+      focusOnInput();
+    }
+  };
+
+  const hundleDeleteTodo = (todosId: number[]) => {
+    setSelectedTodosId(todosId);
+    todosId.map(todoId => (
```

Comment:
use promise.all with map
[here](https://betterprogramming.pub/how-to-use-async-await-with-map-in-js-5059043564e0) more info